src/main/java/unicash/model/transaction/predicates/TransactionCategoryContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,60 @@
+package unicash.model.transaction.predicates;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import unicash.commons.util.StringUtil;
+import unicash.commons.util.ToStringBuilder;
+import unicash.model.transaction.Transaction;
+
+/**
+ * Tests that a {@code Transactions}'s {@code UniqueCategoryList} matches any of
+ * the keywords given.
+ */
+public class TransactionCategoryContainsKeywordsPredicate
+        implements Predicate<Transaction> {
+    private final List<String> keywords;
+
+    public TransactionCategoryContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    /* A nested anyMatch compares each keyword for each category in the categoryList.
+     * This removes any String dependency entirely so the keyword "od,so" now will not
+     * match something like "food,social" compared to if the string representation
+     * of the categoryList was returned.
+     */
+    @Override
+    public boolean test(Transaction transaction) {
+        return keywords
+                .stream().anyMatch(keyword -> transaction
+                        .getCategories().joinCategoriesAsList()
+                                .stream().anyMatch(category ->
+                                        StringUtil.containsSubstringIgnoreCase(
+                                                        category.toString(), keyword)
+                                )
+                );
```

Comment:
```suggestion
        return keywords
                .stream()
                .anyMatch(keyword -> transaction
                        .getCategories()
                        .joinCategoriesAsList()
                        .stream()
                        .anyMatch(category -> 
                            StringUtil.containsSubstringIgnoreCase(
                                category.toString(), 
                                keyword
                            )
                        )
                );
```

The current indentation is quite hard to read

Reply:
Fixed